component/networkpolicy.libsonnet

```diff
@@ -0,0 +1,96 @@
+local com = import 'lib/commodore.libjsonnet';
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+
+local inv = kap.inventory();
+local params = inv.parameters.argocd;
+local isOpenshift = std.member([ 'openshift4', 'oke' ], params.distribution);
+
+[
+  kube.NetworkPolicy('argocd-allow-same-namespace') {
+    spec: {
+      ingress: [
+        {
+          from: [
+            {
+              podSelector: {},
+            },
+          ],
+        },
+      ],
+      policyTypes: [
+        'Ingress',
+      ],
+      podSelector: {},
+    },
+  },
+  kube.NetworkPolicy('argocd-allow-operator') {
```

Comment:
Is this policy actually necessary? Afaict, we don't have it currently and everything seems to work just fine.

Reply:
I thought currently we don't have any access control. But I did go out and test the whole setup without this policy, and it seems to indeed not be required, so I shall remove it.

Reply:
> I thought currently we don't have any access control.

To clarify: we currently have namespace isolation enabled on VSHN-managed OCP4. This change is primarily intended to restrict access to the ArgoCD components (in particular Redis) on other installations.